lib/Dialect/Seq/SeqPasses.cpp

```diff
@@ -43,15 +43,21 @@ struct CompRegLower : public OpConversionPattern<CompRegOp> {
     Location loc = reg.getLoc();
 
     StringAttr name = reg.nameAttr();
-    if (!name)
+    if (reg.name().empty())
```

Comment:
You can delete this `if` entirely.  No need to synthesize a _compreg name since reg allows empty names.

Reply:
I get that there's no need, but is there anything wrong with anything lowered from compreg being named `_compreg_*` instead of `_T_*`? It's strictly more information and could help us debug passes.

Reply:
There is nothing locally wrong with this, but it leads to global problems.  The same name could be produced by whatever is creating the compreg upstream so this isn't necessary.  However, the absence of a name is much easier to notice when it is missing, rather than being filled in automagically by stuff like this.  As such, it is best to have a consistent policy where we don't fill in compiler generated things.

Reply:
Good point. 4730fe83618c1e19d111800fec9c0e4aa03e4539